apps/sms/views/shared/js/activity_handler.js

```diff
@@ -348,11 +346,11 @@ var ActivityHandler = {
           // 'mms message' in the field.
 
           if (needManualRetrieve) {
-            return formatValue('notDownloaded-title');
+            return 'notDownloaded-title';
           }
 
           if (message.subject) {
-            return Promise.resolve(message.subject);
+            return {raw: message.subject};
           }
 
           return SMIL.parse(message).then((slideArray) => {
```

Comment:
this line will return a promise so you need to return a promise for the previous cases as well